pkg/rbac/dbAvailability.go

```diff
@@ -0,0 +1,21 @@
+package rbac
+
+import (
+	"net/http"
+
+	"k8s.io/klog/v2"
+)
+
+func (c *Cache) CheckDBAvailability(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// if postgres db is not setup, return error
+		if !c.dbConnInitialized {
+			http.Error(w, "Unable to establish connection with database.", http.StatusServiceUnavailable)
```

Comment:
Let's log this error too.
```suggestion
			klog.Warn("Unable to handle request because we couldn't establish connection with database.") 
			http.Error(w, "Unable to establish connection with database.", http.StatusServiceUnavailable)
```